### DISCUSSION TITLE
Action Timeout

### DISCUSSION THREAD

**Discussion by komalphadatare** [StackStorm/st2#5514] [Q&A] [answered]
Hi,
In a workflow if any action is getting timed out i want to add a transition similar to success or failed.
I tried using jinja tags eg: <% timeout() %> similar to succeeded and failed but no luck.
we tried above approach for action type core.remote , core.http

**Comment by nzlosh** [comment, marked as answer]
The below example shows the output of an action that timed out:

```
st2 run core.local cmd='sleep 10' timeout=5
...
id: 61c2ddf2d884a3a0c05c3f8b
action.ref: core.local
context.user: st2admin
parameters: 
  cmd: sleep 10
  timeout: 5
status: timeout
start_timestamp: Wed, 22 Dec 2021 08:12:34 UTC
end_timestamp: Wed, 22 Dec 2021 08:12:40 UTC
result: 
  error: Action failed to complete in 5 seconds
  failed: true
  return_code: -9
  stderr: ''
  stdout: ''
  succeeded: false
```
The `status` keyword indicates the action timed out.  You can write a condtition in the `next` section to decide how the workflow should proceed.
```
      - when: <% task().status = "timeout" %>
        do: action_to_run_when_this_action_times_out
```
